client/src/main/java/com/tigrisdata/db/client/StreamEvent.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2022 Tigris Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tigrisdata.db.client;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tigrisdata.db.api.v1.grpc.Api;
+
+public class StreamEvent {
+  private byte[] txId;
+  private String collection;
+  private String op;
+  private byte[] key;
+  private byte[] lKey;
+  private byte[] rKey;
+  private JsonNode data;
+  private boolean last;
+
+  public static StreamEvent from(Api.StreamEvent apiEvent, ObjectMapper objectMapper)
+      throws IOException {
+    StreamEvent event = new StreamEvent();
+    event.setTxId(apiEvent.getTxId().toByteArray());
+    event.setCollection(apiEvent.getCollection());
+    event.setOp(apiEvent.getOp());
+    event.setKey(apiEvent.getKey().toByteArray());
+    event.setLKey(apiEvent.getLkey().toByteArray());
+    event.setRKey(apiEvent.getRkey().toByteArray());
+    event.setData(objectMapper.readTree(apiEvent.getData().toByteArray()));
+    event.setLast(apiEvent.getLast());
```

Comment:
while we control the call to constructor. Can we simply make this class immutable ?

Reply:
We could, it just seemed more like a POJO that might be desirable to easily serialize